src/App.tsx

```diff
@@ -23,47 +24,43 @@ function App() {
     <>
       <ToastContainer />
       <Routes>
-        {
-          isLoggedIn 
-          ? (
-            <Route path="/" element={<LoggedInLayout />}>
-              <Route path="/" element={<HomePage />} />
-              <Route path="/lessons" element={<LessonsPage />} />
-              <Route path="/lessons/:lessonID" element={<LessonDetailPage />} />
-              <Route path="/subject" element={<SubjectPage />} />
-              <Route path="/subject/:subjectCode" element={<SubjectDetailPage />} />
-              <Route path="/classes" element={<StudentClassPage />} />
+        {isLoggedIn ? (
+          <Route path="/" element={<LoggedInLayout />}>
+            <Route path="" element={<HomePage />} />
```

Comment:
Eu sei que já tava assim, mas eu aprendi que o certo do reactDOM qnd vc quer q um compnente renderize em cima de outro, que é o caso aqui, é você omitir o path e usar a prorpiedade `index`:
```suggestion
            <Route index element={<HomePage />} />
```